apps/admin-ui/src/spa/reservations/[id]/ReservationKeylessEntrySection.tsx

```diff
@@ -0,0 +1,251 @@
+}>`
+  gap: var(--spacing-s);
+  padding: var(--spacing-m);
+  background: var(--color-black-5);
+  justify-content: space-between;
+  display: grid;
+
+  ${({ $isRecurring }) =>
+    $isRecurring
+      ? css`
+          grid-template-columns: repeat(2, 1fr);
+          @media (min-width: ${breakpoints.l}) {
+            grid-template-columns: repeat(6, 1fr);
+          }
+        `
+      : css`
+          grid-template-columns: repeat(3, 1fr);
+          @media (min-width: ${breakpoints.s}) {
+            grid-template-columns: repeat(4, 1fr);
+          }
+        `};
+`;
+
+const SingleButtonContainer = styled(ButtonContainer)<{
+  $isRecurring?: boolean;
+}>`
+  ${({ $isRecurring }) =>
+    $isRecurring
+      ? css`
+          @media (max-width: ${breakpoints.l}) {
+            grid-column: 1 / span 2;
+          }
+          @media (min-width: ${breakpoints.l}) {
+            width: auto;
+          }
+        `
+      : css`
+          @media (max-width: ${breakpoints.s}) {
+            grid-column: 1 / span 3;
+          }
+          @media (min-width: ${breakpoints.s}) {
+            width: auto;
+          }
+        `};
+
+  justify-content: flex-end;
+`;
+
+export function ReservationKeylessEntry({
+  reservation,
+  onSuccess,
+}: Readonly<{
+  reservation: ReservationType;
+  onSuccess: () => void;
+}>) {
+  const { t } = useTranslation();
+
+  return (
+    <Accordion
+      id="reservation__access-type"
+      heading={t("RequestedReservation.keylessEntry")}
+      initiallyOpen={false}
+    >
+      <div>
+        <SummaryHorizontal $isRecurring={!!reservation.recurringReservation}>
+          <DataWrapper label={t("RequestedReservation.accessCodeLabel")}>
+            {reservation.pindoraInfo?.accessCode ?? "-"}
+          </DataWrapper>
+
+          <DataWrapper label={t("RequestedReservation.accessCodeStatusLabel")}>
+            {reservation.pindoraInfo?.accessCodeIsActive
+              ? t("RequestedReservation.accessCodeStatusActive")
+              : t("RequestedReservation.accessCodeStatusInactive")}
+            {reservation.pindoraInfo?.accessCodeIsActive !==
+              reservation.accessCodeShouldBeActive && <IconAlertCircleFill />}
+          </DataWrapper>
+
+          {reservation.recurringReservation && (
+            <>
+              <DataWrapper label={t("common.startingDate")}>
+                {reservation.recurringReservation?.beginDate
+                  ? formatDate(reservation.recurringReservation.beginDate)
+                  : "-"}
+              </DataWrapper>
+              <DataWrapper label={t("common.endingDate")}>
+                {reservation.recurringReservation?.endDate
+                  ? formatDate(reservation.recurringReservation.endDate)
+                  : "-"}
+              </DataWrapper>
+            </>
+          )}
+
+          <DataWrapper
+            label={t("RequestedReservation.accessCodeValidityLabel")}
+          >
+            {reservation.pindoraInfo
```

Comment:
In the recurring case, since `pindoraInfo.accessCodeValidity` is an array of 1-N occurrences, we shall display the time of the currently ongoing or next (future) one contained in it (cf. tooltip in Figma). Or lacking either, the last (past) one.